Upstream CI and fixed smoketest

pleimer opened the pull request:
add Jenkins pipeline and fix smoketest for stf 1.2

pleimer commented:
test

leifmadsen commented on Jenkinsfile:
```diff
@@ -0,0 +1,153 @@
+  transports:
+    qdr:
+      enabled: true
+      deployment_size: 1
```
In the spec you should be using camelCase not snake_case. The snake_case conversion happens at the Operator level.

leifmadsen commented on Jenkinsfile:
```diff
@@ -0,0 +1,153 @@
+                                colorized: true,
+                                extraVars: [
+                                    "namespace": namespace,
+                                    "__deploy_stf": "false",
```
Huh this works? I'd think you'd need STF deployed in order for the validation and smoke tests to pass?

pleimer replied:
STF is deployed in the next step of the Jenkins file. I needed to use a custom manifest to minimize the ES footprint to get reliable deployments on Quicklab, so I deploy it straight from Jenkinsfile

csibbitt replied:
So this call to run-ci is actually more like build-containers-for-ci? I kinda get that from the stage name, but a one-line comment above L112 saying something like "Builds containers but does not actually run CI" might be helpful

leifmadsen commented on Jenkinsfile:
```diff
@@ -0,0 +1,153 @@
+                stage('Cleanup') {
+                    openshift.withCluster(){
+                        openshift.selector("project/${namespace}").delete()
+                        if ( stages_failed ) { currentBuild.result = 'FAILUR' }
```
```suggestion
                        if ( stages_failed ) { currentBuild.result = 'FAILURE' }
```

leifmadsen commented on Jenkinsfile:
```diff
@@ -0,0 +1,153 @@
+                        openshift.selector("project/${namespace}").delete()
+                        if ( stages_failed ) { currentBuild.result = 'FAILUR' }
+                    }
+                    if ( stages_failed ) { curreentBuild.result = 'FAILURE' }
```
```suggestion
                    if ( stages_failed ) { currentBuild.result = 'FAILURE' }
```

leifmadsen commented on tests/smoketest/minimal-collectd.conf.template:
```diff
@@ -12,7 +12,7 @@ LoadPlugin amqp1
   <Transport "name">
     Host "default-interconnect"
-    Port "5672"
+    Port "5671"
```
I thought 5671 was for the internal QDR connections?

leifmadsen replied:
Yea I didn't look through this and it was probably just late when I looked at this. I believe you :D

csibbitt replied:
> I thought 5671 was for the internal QDR connections?

According to our wiki page: 

```
5671 | edge, Anonymous SASL, SSL (profile: openstack) - Used for connections from OSP
5672 | edge, no auth, no ssl - Used by legacy SG
```

However AFAICT this port is being used without SSL in these tests. Is our wiki wrong? I don't have a live system in front of me to check.

leifmadsen commented on tests/smoketest/smoketest.sh:
```diff
@@ -6,25 +6,35 @@
 #   * gnu sed
 #
-# Usage: ./smoketest.sh [NUMCLOUDS]
+# Usage: NUMCLOUDS=<num clouds> OCP_PROJECT=<project> ./smoketest.sh [NUMCLOUDS]
```
I'm confused by this change. The script uses a parameter passed to it with a default of 1. Seems like you're basically just overriding that? See line 16 for where it is set. I'm not sure this is necessary? (I actually like env param better than the argument method fwiw.)

pleimer replied:
That looks like a relic from the backport. Someone else must have made that change. Not sure why that was ever changed

pleimer replied:
Looks like Chris made this change over a year ago when he imported changes from the old repo
```
commit f34637b5d45e59b21c21d81f092286b9955cfc75 (HEAD)
Author: Chris Sibbitt <csibbitt@redhat.com>
Date:   Fri Jan 24 17:10:27 2020 -0500

    Importing content from deprecated telemetry-framework repo
    
    * From an up to date master of both
    ```
    $ cp -a ../telemetry-framework/deploy/ .
    $ rm deploy/configs/es-workaround.bash  # We don't need this anymore
    $ cp -a ../telemetry-framework/tests/ .
    $ git add -A
    ```
    * Merged important content from README.md
    
    History before this point may be available at
    https://github.com/redhat-service-assurance/telemetry-framework
```
 I'm willing to switch it back over, but this is a change that will also need to be made in 1.3 and master. I'll include it in a new patch that I propose there and backport to here again

leifmadsen commented on tests/smoketest/smoketest.sh:
```diff
@@ -82,8 +93,7 @@ oc logs "$(oc get pod -l "smart-gateway=default-cloud1-ceil-event" -o jsonpath='
 echo "*** [INFO] Logs from smart gateway operator..."
-oc logs "$(oc get pod -l app=smart-gateway-operator -o jsonpath='{.items[0].metadata.name}')" -c ansible
-echo
+oc logs "$(oc get pod -l app=smart-gateway-operator -o jsonpath='{.items[0].metadata.name}')" -c ansible echo
```
Do you mean for a `;` to separate `ansible` and `echo` here?

pleimer replied:
nope!

csibbitt replied:
This doesn't seem right. I'd expect it to return: `error: only one of -c or an inline [CONTAINER] arg is allowed`

pleimer replied:
`-c ansible` is specifyng the container from which to print logs. Also I did mean to put a semicolon there

pleimer commented on .jenkins/agent/Dockerfile:
```diff
@@ -0,0 +1,7 @@
+FROM quay.io/openshift/origin-jenkins-agent-base:latest
+
+COPY install-sdk.sh /install-sdk.sh
+RUN ./install-sdk.sh
```
bummer I am going to have to change that in the 1.3 and master as well

csibbitt commented on Jenkinsfile:
```diff
@@ -0,0 +1,153 @@
+                      memory: 4Gi
+                    requests:
+                      cpu: '1'
+                      memory: 2Gi
```
Is this change from 4G to 2G the only difference with your manifest vs. the original, or is there something else as well?

We really need a better way to make targeted changes like this.

pleimer replied:
Yeah that's the only reason I don't deploy stf from the run-ci playbook. 2G consistently deploys on quicklab, 4G does not

csibbitt commented on tests/smoketest/smoketest.sh:
```diff
@@ -37,22 +46,23 @@ for NAME in "${CLOUDNAMES[@]}"; do
 
 # Trying to find a less brittle test than a timeout
-JOB_TIMEOUT=500s
+JOB_TIMEOUT=300s
```
On the one hand, shorter timeout is better. On the other, the comment on the line above includes the word "brittle". So I dunno.....

pleimer replied:
300s was the original value, then at some point we made it longer because we thought it wasn't long enough but the reality was that the tripleo containers were not able to be pulled in because of the Dockerhub rate limiting

csibbitt commented on tests/smoketest/smoketest.sh:
```diff
@@ -97,6 +106,15 @@ echo "*** [INFO] Logs from snmp webhook..."
 echo
 
+echo "*** [INFO] Cleanup resources..."
+oc delete job/stf-smoketest-${NAME}
```
It would be nice to have an argument to skip this since I think it removes the failed pods that you might want to examine.

pleimer replied:
That's a good idea

csibbitt replied:
Right, manual testing (or temporary no-cleanup on the CI server to permit examination) is what I'm thinking of. I often run this script manually after doing a deployment and would like to investigate if it fails.

pleimer commented:
test

pleimer commented:
test

pleimer commented:
test

pleimer commented:
Getting the following error when attempting to deploy STF-1.2 on OCP4.7. This must have something to do with an incompatability with 4.7

```

TASK [stf-run-ci : Load Smart Gateway Operator RBAC] ***************************
changed: [localhost] => (item=service_account.yaml)
changed: [localhost] => (item=role.yaml)
changed: [localhost] => (item=role_binding.yaml)
failed: [localhost] (item=olm-catalog/smart-gateway-operator/manifests/smartgateway.infra.watch_smartgateways_crd.yaml) => {"ansible_loop_var": "item", "changed": true, "cmd": ["oc", "apply", "-f", "working/smart-gateway-operator/deploy/olm-catalog/smart-gateway-operator/manifests/smartgateway.infra.watch_smartgateways_crd.yaml", "-n", "pr-243-45"], "delta": "0:00:01.611777", "end": "2021-08-19 20:13:07.758549", "item": "olm-catalog/smart-gateway-operator/manifests/smartgateway.infra.watch_smartgateways_crd.yaml", "msg": "non-zero return code", "rc": 1, "start": "2021-08-19 20:13:06.146772", "stderr": "W0819 20:13:07.126490    1084 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition\nW0819 20:13:07.749843    1084 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition\nThe CustomResourceDefinition \"smartgateways.smartgateway.infra.watch\" is invalid: status.storedVersions[0]: Invalid value: \"v2\": must appear in spec.versions", "stderr_lines": ["W0819 20:13:07.126490    1084 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition", "W0819 20:13:07.749843    1084 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition", "The CustomResourceDefinition \"smartgateways.smartgateway.infra.watch\" is invalid: status.storedVersions[0]: Invalid value: \"v2\": must appear in spec.versions"], "stdout": "", "stdout_lines": []}
```

leifmadsen commented:
recheck

leifmadsen commented:
I suspect this will fail, and we'll need to adjust the `Jenkinsfile` like we did for `master` and `stable-1.3`.

pleimer commented:
> I suspect this will fail, and we'll need to adjust the `Jenkinsfile` like we did for `master` and `stable-1.3`.

We aren't going to be able to make this work unless we use 4.7 for our CI cluster or else update the stf-1.2 APIs

leifmadsen commented:
> > I suspect this will fail, and we'll need to adjust the `Jenkinsfile` like we did for `master` and `stable-1.3`.
> 
> We aren't going to be able to make this work unless we use 4.7 for our CI cluster or else update the stf-1.2 APIs

Ah right, or create another 4.6 cluster for STF 1.2.

I think since STF 1.4 is basically gonna be due in late Q4, early Q1, which would make STF 1.2 EOL, I'm not sure it's worth the effort to add this for upstream... (I say that now, but we don't seem to have a lot of STF 1.2 backports at this point).

pleimer commented:
Agreed, I say we continue with this effort only if it proves necessary

leifmadsen commented:
> Agreed, I say we continue with this effort only if it proves necessary

We can probably just close this PR out then.

pleimer commented:
This PR also had changes for the smoketest that were useful in testing 1.2 CVEs, but I'll extrapolate those changes elsewhere.
